app/controllers/application_controller.rb

```diff
@@ -1,10 +1,150 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+   register Sinatra::ActiveRecordExtension
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect("/tweets")
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if !params["username"].empty? && !params["email"].empty? && !params["password"].empty?
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect("/tweets")
+    else
+     redirect("/signup")
+   end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = current_user
+      erb :'/users/show'
+    else
+      redirect("/login")
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = current_user
+      erb :'/tweets/create_tweet'
+    else
+      redirect("/login")
+    end
+  end
+
+  post '/tweets' do
+    if logged_in? && !params["content"].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.